### PR TITLE
[BUG] fix `FeatureUnion` test failure mk4

### PR DESCRIPTION
**Pull request by fkiraly**
Another attempted fix for #1662 which PR #1665, #1701 did not fix. Hopefully the definitive fix now.

This fix completely rewrites the `FeatureUnion` instead of inheriting from the `sklearn` one.
With the new transformers refactor, we can simply use `pandas` functionality to merge/concatenate data frames internally, while the framework does the rest.

Minor complication:
With the transformer refactor, the "row-wise transformer" have become obsolete, since vectorization (to panel) is performed automatically whenever such a data container is encountered.
Some tests not directly related to the `FeatureUnion` but also using `FeatureUnion` were still using them, so I `xfailed` them for the time being, until the `RowTransformer`s are entirely removed, see issue #2179.

### CODE REVIEW

**Comment by fkiraly** [comment]
@chrisholder, thanks for reviewing #1701!
I noticed that I hadn't pushed the most recent and hopefully final attempt to fix this.
Pinging you since you recently looked at the bugfix.

Also FYI @lmmentel, as we've discussed earlier on slack, this is the branch now as PR.

File: sktime/series_as_features/compose/_pipeline.py [hunk 1]
```diff
@@ -35,55 +37,126 @@ class FeatureUnion(_FeatureUnion, _PanelToPanelTransformer):
...
+        "transform-returns-same-time-index": False,
+        "skip-inverse-transform": False,
+    }
+
     def __init__(
         self,
         transformer_list,
         n_jobs=None,
         transformer_weights=None,
         preserve_dataframe=True,
```
Comment (chrisholder) [inline sktime/series_as_features/compose/_pipeline.py:67]: Maybe have a deprecated message? I dont know could get spammy?

File: sktime/series_as_features/compose/_pipeline.py [hunk 2]
```diff
@@ -35,55 +37,126 @@ class FeatureUnion(_FeatureUnion, _PanelToPanelTransformer):
...
-        else:
-            return np.hstack(Xs)
+    @classmethod
+    def get_test_params(cls):
+        """Test parameters for FeatureUnion."""
+        from sklearn.preprocessing import StandardScaler
+
+        SERIES_TO_SERIES_TRANSFORMER = StandardScaler()
+
+        from sktime.transformations.panel.compose import SeriesToSeriesRowTransformer
```
Comment (chrisholder) [inline sktime/series_as_features/compose/_pipeline.py:141]: Move to top :)

**Review by chrisholder — CHANGES_REQUESTED** [review]

File: sktime/series_as_features/compose/_pipeline.py (hunk 1 above)
Comment (fkiraly) [reply sktime/series_as_features/compose/_pipeline.py:67]: can you kindly explain for what?
I don't think anything deprecates here? Still fully downwards compatible?

File: sktime/series_as_features/compose/_pipeline.py (hunk 2 above)
Comment (fkiraly) [reply sktime/series_as_features/compose/_pipeline.py:141]: Per design, imports for the test instances should be inside the testing methods.
Perhaps I should clarify this in the extension template?

Comment (fkiraly) [reply sktime/series_as_features/compose/_pipeline.py:141]: I made a PR that clarifies this in the extension template:
https://github.com/alan-turing-institute/sktime/pull/1811

**Comment by lmmentel** [comment]
It seems that quite a few tests on linux are still failing:

```
FAILED classification/compose/tests/test_TimeSeriesForestClassifier.py::test_equivalent_model_specifications[1-1]
FAILED classification/compose/tests/test_TimeSeriesForestClassifier.py::test_equivalent_model_specifications[3-1]
FAILED tests/test_all_estimators.py::test_fit_idempotent[FeatureUnion] - Valu...
FAILED tests/test_all_estimators.py::test_methods_do_not_change_state[FeatureUnion]
FAILED tests/test_all_estimators.py::test_methods_have_no_side_effects[FeatureUnion]
FAILED tests/test_all_estimators.py::test_persistence_via_pickle[FeatureUnion]
FAILED tests/test_all_estimators.py::test_multiprocessing_idempotent[FeatureUnion]
FAILED transformations/panel/summarize/tests/test_RandomIntervalFeatureExtractor.py::test_different_pipelines
```

But not the test related to hyper params :slightly_smiling_face:

**Comment by fkiraly** [comment]
Yes, this will probably require #1795 to work.

File: sktime/series_as_features/compose/_pipeline.py (hunk 1 above)
Comment (chrisholder) [reply sktime/series_as_features/compose/_pipeline.py:67]: In the Parameter comment it has 'preserve_dataframe : bool - deprecated'

Comment (fkiraly) [reply sktime/series_as_features/compose/_pipeline.py:67]: ah! Yes, indeed, that makes sense.
Or, I should perhaps better think about how to make it downwards compatible.

Comment (fkiraly) [reply sktime/series_as_features/compose/_pipeline.py:68]: added a deprecation message. The class itself should still be downwards compatible

**Comment by fkiraly** [comment]
@chrisholder, @lmmentel, after the transformers refactor, the featureunion fix works with some minor changes, kindly have a look.

**Comment by fkiraly** [comment]
comments addressed, no activity for >14 days, merging

**Pull request merged**